### ISSUE TITLE
Some Prometheus metrics still missing for Kubernetes deployment

### ISSUE THREAD

**Issue by jldiaz-uniovi** [microservices-demo/microservices-demo#857]
Issue #850 was closed after @rodrmartinez suggested to add the metadata `prometheus.io/scrape: 'true'` to all services.

This in fact solves part of the problem. Now the grafana dashboards "Sock-Shop Analytics" and "Sock-Shop Performances" are populated. However, the dashboards "Sock-Shop Resources" and "Kubernetes Pod resources" still show no data.

After some research, I solved the problem by making the following changes:

* In file `deploy/kubernetes/manifests-monitoring/09-prometheus-dep.yaml` the line `serviceAccount: prometheus` has to be changed to `serviceAccountName: prometheus`
* In file `deploy/kubernetes/manifests-monitoring/02-prometheus-cr.yaml` the resource `nodes/proxy` has to be added to the list of resources for the second rule. I.e., this file has to be like this:

```yaml
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: prometheus
  labels:
    app: prometheus
rules:
- apiGroups:
  - networking.k8s.io
  resources:
  - ingresses
  verbs:
  - get
  - list
  - watch
- apiGroups: [""] # "" indicates the core API group
  resources:
  - nodes
  - nodes/metrics
  - nodes/proxy
  - services
  - endpoints
  - pods
  verbs:
  - get
  - list
  - watch
- nonResourceURLs:
  - /metrics
  verbs:
  - get
```

After these changes, all Grafana dashboards show data now.

**Comment by elieserr** [comment]
Hi @jldiaz-uniovi would you like to open a PR with your suggested solution?

**Comment by rodrmartinez** [comment]
Solved by  #858
